Sources/StreamChat/ChatClient.swift

```diff
@@ -379,6 +382,7 @@ public class ChatClient {
         userInfo: UserInfo,
         completion: ((Error?) -> Void)? = nil
     ) {
+        connectionRecoveryHandler?.start()
```

Comment:
@nuno-vieira @martinmitrevski Hello, I found that this code is causing an issue, so I'm leaving a comment.

`connectGuestUser` calls logOut first. As a result, ConnectionRecoveryHandler.stop is invoked, causing the monitoring of appDidBecomeActive and appDidEnterBackground to stop. This leads to the issue where "the WebSocket does not reconnect when the app resumes from the background."

Reply:
I posted issue https://github.com/GetStream/stream-chat-swift/issues/3338

Reply:
Hi @masarusanjp, thanks for spotting this issue!

The actual issue is that when reconnecting, logOut should not be happening, so the logic should be changed here. Stopping the recovery handler on logout is something that we want, but it caused this unwanted side effect due to the fact that logOutFirst in guest users is not properly implemented, it seems.